packages/frontend/src/connectors/readonly/index.ts

```diff
@@ -2,14 +2,33 @@ import { JsonRpcProvider } from "@ethersproject/providers";
 import { Connector } from "wagmi";
 import { Address, Chain, ConnectorData, normalizeChainId } from "@wagmi/core";
 import { constants } from "ethers";
+import { ChainId } from "@carrot-kpi/sdk";
 
 export const READ_ONLY_CONNECTOR_ID = "readonly";
+const DEFAULT_CHAIN_ID = ChainId.GNOSIS; // 100
 
 export type InjectedConnectorOptions = {
     /** Name of connector */
     name?: string | ((detectedName: string | string[]) => string);
 };
 
+const getChainIdFromlocalStorage = () => {
+    const wagmiStore = localStorage.getItem("wagmi.store");
+    if (!wagmiStore) return DEFAULT_CHAIN_ID;
+    const parsedStore = JSON.parse(wagmiStore);
```

Comment:
Should we add a type for the parsed wagmi state object?

Reply:
added `wagmiStoreObj` interface

Reply:
added `wagmiStoreObj` interface

Reply:
fixed on e0b89de42772c6efefdb918a999147a0a12a0c4f